interfaces/udev/spec.go

```diff
@@ -163,3 +164,24 @@ func (spec *Specification) AddPermanentSlot(iface interfaces.Interface, slot *sn
 	}
 	return nil
 }
+
+// Informs ReloadRules() to also do 'udevadm trigger <subsystem specific>'.
+// IMPORTANT: because there is currently no way to call TriggerSubsystem during
+// interface disconnect, TriggerSubsystem() should typically only by used in
+// UDevPermanentSlot since the rules are permanent until the snap is removed.
+func (spec *Specification) TriggerSubsystem(subsystem string) {
+	if subsystem == "" {
+		return
+	}
+
+	for _, elem := range spec.udevadmSubsystemTriggers {
```

Comment:
You could perhaps use `strutil.ListContains` instead.
```
if subsystem != "" && !strutil.ListContains(spec.udevadmSubsystemTriggers, subsystem) {
    spec.udevadmSubsystemTriggers = append(spec.udevadmSubsystemTriggers, subsystem)
}
```

Offtopic: the struct field name is mouthful, could we just call it `triggers`?

Reply:
I like the specificity of it. It is for triggering a particular subsystem when calling udevadm. Trigger doesn't mean anything in particular when talking about 'udev' generally and it is specific to the udevadm implementation. I can change this if you feel strongly about it.

Reply:
No, that's fine.